src/contracts/FOXy.sol

```diff
@@ -4,247 +4,7 @@ pragma solidity 0.8.11;
-     * invalid opcode to revert (consuming all remaining gas).
-     *
-     * Requirements:
-     *
-     * - The divisor cannot be zero.
-     */
-    function mod(uint256 a, uint256 b) internal pure returns (uint256) {
-        return mod(a, b, "SafeMath: modulo by zero");
-    }
-
-    /**
-     * @dev Returns the remainder of dividing two unsigned integers. (unsigned integer modulo),
-     * Reverts with custom message when dividing by zero.
-     *
-     * Counterpart to Solidity's `%` operator. This function uses a `revert`
-     * opcode (which leaves remaining gas untouched) while Solidity uses an
-     * invalid opcode to revert (consuming all remaining gas).
-     *
-     * Requirements:
-     *
-     * - The divisor cannot be zero.
-     */
-    function mod(
-        uint256 a,
-        uint256 b,
-        string memory errorMessage
-    ) internal pure returns (uint256) {
-        require(b != 0, errorMessage);
-        return a % b;
-    }
-
-    // babylonian method (https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Babylonian_method)
-    function sqrrt(uint256 a) internal pure returns (uint256 c) {
-        if (a > 3) {
-            c = a;
-            uint256 b = add(div(a, 2), 1);
-            while (b < c) {
-                c = b;
-                b = div(add(div(a, b), b), 2);
-            }
-        } else if (a != 0) {
-            c = 1;
-        }
-    }
-
-    /*
-     * Expects percentage to be trailed by 00,
-     */
-    function percentageAmount(uint256 total_, uint8 percentage_)
-        internal
-        pure
-        returns (uint256 percentAmount_)
-    {
-        return div(mul(total_, percentage_), 1000);
-    }
-
-    /*
-     * Expects percentage to be trailed by 00,
-     */
-    function substractPercentage(uint256 total_, uint8 percentageToSub_)
-        internal
-        pure
-        returns (uint256 result_)
-    {
-        return sub(total_, div(mul(total_, percentageToSub_), 1000));
-    }
-
-    function percentageOfTotal(uint256 part_, uint256 total_)
-        internal
-        pure
-        returns (uint256 percent_)
-    {
-        return div(mul(part_, 100), total_);
-    }
-
-    /**
-     * Taken from Hypersonic https://github.com/M2629/HyperSonic/blob/main/Math.sol
-     * @dev Returns the average of two numbers. The result is rounded towards
-     * zero.
-     */
-    function average(uint256 a, uint256 b) internal pure returns (uint256) {
-        // (a + b) / 2 can overflow, so we distribute
-        return (a / 2) + (b / 2) + (((a % 2) + (b % 2)) / 2);
-    }
-
-    function quadraticPricing(uint256 payment_, uint256 multiplier_)
-        internal
-        pure
-        returns (uint256)
-    {
-        return sqrrt(mul(multiplier_, payment_));
-    }
-
-    function bondingCurve(uint256 supply_, uint256 multiplier_)
-        internal
-        pure
-        returns (uint256)
-    {
-        return mul(multiplier_, supply_);
-    }
-}
-
 abstract contract ERC20 is IERC20 {
```

Comment:
Future TODO: see if we can just pull in one of  OZ's erc20 implementations

Reply:
I originally tried to do this, but due to accessing _totalSupply it was erroring out and decided to skip this for now.  I will add a ticket to do this because it will cause less of a headache